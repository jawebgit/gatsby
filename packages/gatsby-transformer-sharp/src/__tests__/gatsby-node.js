const Promise = require(`bluebird`)

const { onNodeCreate } = require(`../gatsby-node`)

describe(`Process image nodes correctly`, () => {
  it(`correctly creates an ImageSharp node from a file image node`, async () => {
    const node = {
      id: `whatever`,
      contentDigest: `whatever`,
      extension: `png`,
      children: [],
    }
    const createNode = jest.fn()
    const updateNode = jest.fn()
    const boundActionCreators = { createNode, updateNode }

    await onNodeCreate({
      node,
      boundActionCreators,
    }).then(() => {
      expect(createNode.mock.calls).toMatchSnapshot()
      expect(updateNode.mock.calls).toMatchSnapshot()
      expect(createNode).toHaveBeenCalledTimes(1)
      expect(updateNode).toHaveBeenCalledTimes(1)
    })
  })
})
