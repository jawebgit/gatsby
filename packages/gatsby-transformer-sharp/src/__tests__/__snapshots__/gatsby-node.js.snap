// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process image nodes correctly correctly creates an ImageSharp node from a file image node 1`] = `Array []`;

exports[`Process image nodes correctly correctly creates an ImageSharp node from a file image node 2`] = `Array []`;
